src/FundSourceApprover/FundSourceAllowerAdmin.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.16;
+
+import "openzeppelin-contracts/contracts/access/Ownable.sol";
+import "openzeppelin-contracts/contracts/utils/Create2.sol";
+import "openzeppelin-contracts/contracts/utils/Address.sol";
+import "./FundSourceAllower.sol";
+
+///@notice Creates FundSourceAllower contracts of which it is the admin.
+contract FundSourceAllowerAdmin is Ownable {
+    bytes32 constant SALT_CONSTANT = keccak256("ARB_AEP");
+    // address to which eth funds in fundSourceAllowers get transfered
+    address public immutable ethDestination;
+
+    // address to which tokens in fundSourceAllowers get transfered
+    address public immutable tokenDestination;
+
+    error NotAContract(address addr);
+
+    event NewFundSourceAlowerCreated(address addr, uint256 indexed chainId);
+    event ApprovedStateSet(bool approved, address allower);
+
+    /// @param _owner initial address with affordances to create FundSourceAllowers and to toggle aprprovals
+    /// @param _ethDestination address to which funds in created FundSourceAllower get transfered
+    /// @param _tokenDestination address to which funds in created Erc20FundSourceAllower get transfered
+
+    constructor(address _owner, address _ethDestination, address _tokenDestination) {
+        _transferOwnership(_owner);
+        ethDestination = _ethDestination;
+        tokenDestination = _tokenDestination;
+    }
+
+    /// @notice create fund source allower
+    /// @param _sourceChainId chain ID of fund source, or other unique identifier (used for bookkeeping)
+    function createFundSourceAllower(uint256 _sourceChainId) external onlyOwner returns (address) {
+        FundSourceAllower allower = new FundSourceAllower{salt: _getSalt(_sourceChainId)}(
```

Comment:
I'm nervous about uniquing on chain id alone since we might get duplicates. Not sure how to handle that

Reply:
getSalt(inboxPath)
where inboxPath := [l2Inbox, l3Inbox, l4Inbox, l5Inbox, …} up to your layer

Reply:
Could be something like getSalt(inbox, parentChainId, chainId)